src/Microsoft.Health.Dicom.Core/Features/CustomTag/CustomTagService.cs

```diff
@@ -47,5 +105,39 @@ public async Task<AddCustomTagResponse> AddCustomTagAsync(IEnumerable<CustomTagE
             // Current solution is synchronouse, no job uri is generated, so always return emtpy.
             return new AddCustomTagResponse(job: string.Empty);
         }
+
+        public async Task DeleteCustomTagAsync(string tagPath, CancellationToken cancellationToken)
+        {
+            DicomTag[] tags;
+            if (!_dicomTagParser.TryParse(tagPath, out tags, supportMultiple: false))
+            {
+                throw new InvalidCustomTagPathException(
+                    string.Format(CultureInfo.InvariantCulture, DicomCoreResource.InvalidCustomTag, tagPath ?? string.Empty));
+            }
+
+            string normalizedPath = tags[0].GetPath();
+
+            IEnumerable<CustomTagEntry> customTagEntries = await _customTagStore.GetCustomTagsAsync(normalizedPath, cancellationToken);
+            if (customTagEntries.Count() == 0)
+            {
+                throw new CustomTagNotFoundException(
+                   string.Format(CultureInfo.InvariantCulture, DicomCoreResource.CustomTagNotFound, tagPath ?? string.Empty));
+            }
+
+            CustomTagEntry customTagEntry = customTagEntries.First();
+
+            await _customTagStore.StartDeleteCustomTagAsync(normalizedPath, cancellationToken);
+            CustomTagIndexType indexType = CustomTagVRAndIndexTypeMapping[customTagEntry.VR];
+            var deleteFunc = indexTypeAndDeleteFunctionMapping[indexType];
+            long affectedRows;
+            do
+            {
+                affectedRows = await deleteFunc.Invoke(normalizedPath, MaxDeleteRecordCount, cancellationToken);
+                await Task.Delay(DelayBetweenDeleteTransaction);
+            }
+            while (affectedRows == MaxDeleteRecordCount);
+
+            await _customTagStore.CompleteDeleteCustomTagAsync(normalizedPath, cancellationToken);
```

Comment:
Presumably with only 1 delete method, you wouldn't need this start/complete concept which seems like it should be handled by the store directly

Reply:
I'm not sure if the MaxDeleteRecordCount is essential not, plan to bring this on today's parking lot.